### DISCUSSION TITLE
Table of Contents like in the Docs

### DISCUSSION THREAD

**Discussion by MarkwinVI** [mantinedev/mantine#1254] [Feature requests]
Hey, would it be possible to add a "Table of Contents" components that depending on where you are on the screen highlights the respective section header? 

Thanks!

**Comment by UltraInstinct05** [comment]
Mantine UI has examples of it. https://ui.mantine.dev/category/toc

**Comment by rtivital** [comment]
As shown above, we already provide TOC components in Mantine UI. We will consider adding use-scroll-spy hook that will handle the logic of highlighting active items.

**Reply by Bladeyy** [reply]
Hi there. Any update on use-scroll-spy hook for TOC?

**Reply by rtivital** [reply]
No updates, it is in the backlog

**Reply by tonydavidcray** [reply]
Just looking into this functionality. Has it progressed through the backlog as yet? Was having a look through the v6.0 branch and thought that the intersections hook may be able to achieve this? [thanks for a wonderful tool]

**Reply by sxflynn** [reply]
You can look at Mantine's doc website to see how they implement it. It's all open source. It depends on you using headers in your parent page and the header names rendering the table of contents. Right now they use a get-headings script to grab that data to feed the component.

**Comment by MarkwinVI** [comment]
Thanks!

This is exactly what I was looking for, I went through the "core" documentation a dozen times to find this - but somehow completely missed the examples page
